File: apps/admin-v2/src/application/use-cases/team/get-team.use-case.ts
import type { UserRepository } from '@/domain/repositories/user.repository';
import type { UserRole } from '@/domain/entities/user';

export class GetTeamUseCase {
  constructor(private repo: UserRepository) {}

  execute(filters?: { role?: UserRole; excludeRole?: UserRole }) {
    return this.repo.getAll(filters);
  }
}

File: apps/admin_mobile/lib/application/use_cases/team/get_team_use_case.dart
import '../../../domain/entities/user.dart';
import '../../../domain/repositories/user_repository.dart';
import '../../../shared/types/paginated_result.dart';

class GetTeamUseCase {
  final UserRepository _repo;
  GetTeamUseCase(this._repo);

  Future<PaginatedResult<User>> call({UserRole? role, UserRole? excludeRole}) =>
      _repo.getAll(role: role, excludeRole: excludeRole);
}
